src/main/resources/assets/admin/common/js/ui/text/PasswordGenerator.ts

```diff
@@ -120,22 +120,31 @@ module api.ui.text {
         }
 
         private assessComplexity(value: string) {
+            const isValid = this.input.isValid();
+
             if (this.complexity) {
                 this.removeClass(this.complexity);
-                this.complexity = undefined;
-            }
-            if (this.isExtreme(value)) {
-                this.complexity = 'extreme';
-            } else if (this.isStrong(value)) {
-                this.complexity = 'strong';
-            } else if (this.isGood(value)) {
-                this.complexity = 'good';
-            } else if (this.isWeak(value)) {
-                this.complexity = 'weak';
+                this.complexity = null;
             }
-            if (this.complexity) {
-                this.addClass(this.complexity);
-                this.getEl().setAttribute('data-i18n', i18n(`field.pswGenerator.complexity.${this.complexity}`));
+
+            this.toggleClass('invalid', !isValid);
+
```

Comment:
Shouldn't we use ``FormItem.INVALID_CLASS`` here as well?

Reply:
We have a plenty of places, where we used `'invalid'` as string literals in the descendants of the `FormItem`. We should do the refactoring there, but only after the release.